js/example_roulette.js

```diff
@@ -0,0 +1,139 @@
+/**
+Picking the frontpage example after the page has loaded leads to reflows
+and re-rendering. Hence, we pick the example as soon as possible
+
+Copyright: 1999-2017 by D Language Foundation
+License:   http://boost.org/LICENSE_1_0.txt, Boost License 1.0
+*/
+
+/**
+Prepares the DOM layout of a runnable code example.
+Sibling nodes for program arguments (.runnable-examples-args)
+and stdin (.runnable-examples-stdin) are detected.
+The parent node has usually the class runnable-examples.
+pre = element containing the code example
+*/
+function buildRunnableExampleLayout(pre) {
+    var parent = pre.parentNode;
+
+    // helper method to construct a button in vanilla JavaScript
+    function createButton(className, value) {
+        var btn = document.createElement("input");
+        btn.type = "button";
+        btn.className = className;
+        btn.value = value;
+        return btn;
+    }
+
+    // helper method to construct an overlay in vanilla JavaScript
+    // typically overlays are output, args, stdin and the editor instance
+    function createOverlay(className, title, type, value) {
+        var el = document.createElement("div");
+        el.className = className;
+        var titleEl = document.createElement("span");
+        titleEl.className = "d_code_title";
+        titleEl.innerText = title;
+        el.appendChild(titleEl);
+        el.appendChild(document.createElement("br"));
+        var inner = document.createElement(type);
+        inner.className = className;
```

Comment:
Yes, because I tried to make a 1:1 copy of the jQuery version, see e.g. https://github.com/dlang/dlang.org/blob/master/js/run.js#L138

```js
stdin = '<div class="d_code_stdin"><span class="d_code_title">Standard input</span><br>'
        + '<textarea class="d_code_stdin">'+stdin+'</textarea></div>';
```

Imho "translation" and refactoring shouldn't be mixed (even though the temptation is very strong).